apps/channel/routes.coffee

```diff
@@ -1,109 +1,102 @@
-#
-# Routes file that exports route handlers for ease of testing.
-#
-
-Channel = require "../../models/channel"
-ChannelBlocks = require "../../collections/channel_blocks"
-Blocks = require "../../collections/blocks"
-User = require "../../models/user"
-sd = require("sharify").data
-Q = require 'bluebird-q'
-_ = require 'underscore'
-
-fetchChannel = ({slug, authToken = null, url = null, cache = false}) ->
-  dfd = Q.defer()
-  channel = new Channel
-    channel_slug: slug
-
-  blocks = new ChannelBlocks null,
-    channel_slug: slug
-
-  if url
-    channel.url = url
-
-  channel.fetch
-    cache: cache
-    data:
-      auth_token: authToken
-    success: -> 
-      dfd.resolve { channel, blocks }
-    error: (m, err) -> dfd.reject err
-
-  dfd.promise
+Channel = require '../../models/channel'
+graphQL = require '../../lib/graphql'
 
 @channel = (req, res, next) ->
-  slug = req.params.channel_slug
-  authToken = req.user?.get('authentication_token')
-
-  fetchChannel({ slug, authToken })
-    .then ({ channel, blocks }) ->
-      return res.redirect 301, "/#{channel.get('slug')}" if channel.get('class') is 'User'
-      blocks.add channel.get 'contents'
+  token = req.user?.get('authentication_token')
+  options = id: req.params.channel_slug
+  channel = new Channel options
+
+  canQuery = """
+    query channel_can($id: ID!) {
+      channel(id: $id) {
+        can {
+          add_to
+          update
+          follow: update
```

Comment:
Probably makes sense to add this one as a distinct ability as well so we can guard against it on the server.